### PR TITLE
Add OCR Numbers exercise

### PR DESCRIPTION
**Pull request by robtimp**
@masters3d I've attempted to improve on the Ruby version by validating the input has a correct number of lines and columns.

### CODE REVIEW

File: exercises/ocr-numbers/OcrNumbersTest.swift [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+        
+        do {
+            let text =
+                "    _\n" +
+                "  | _\n" +
+                "  ||_\n" +
+                "     "
+            let _ = try OCR(text)
+        } catch OCR.Error.InvalidNumberOfColumns {
+            throwsInvalidNumberOfColumnsError = true
```
Comment (masters3d) [inline exercises/ocr-numbers/OcrNumbersTest.swift]: Can you return here after this assignment?

File: exercises/ocr-numbers/OcrNumbersTest.swift [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+// Apple Swift version 2.1
+
+class OcrNumbersTest: XCTestCase {
+    
+    func testRecognizeZero() {
+        let text =  " _ \n" +
+                    "| |\n" +
+                    "|_|\n" +
+                    "   "
+        XCTAssertEqual("0", try? OCR(text).convert())
```
Comment (masters3d) [inline exercises/ocr-numbers/OcrNumbersTest.swift]: Let's call this 'converted' if is not mutating to start following the new naming guidelines. (Well need to revise the rest of the problems once that comes out. )

Comment (robtimp) [inline exercises/ocr-numbers/OcrNumbersTest.swift]: Ah, good call!

**Comment by robtimp** [comment]
@masters3d Ok, updated

File: exercises/ocr-numbers/OcrNumbersTest.swift [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+        
+        do {
+            let text =
+                "    _  _ \n" +
+                "  | _| _|"
+            let _ = try OCR(text)
+        } catch OCR.Error.InvalidNumberOfLines {
+            throwsInvalidNumberOfLinesError = true
+            return
+        } catch {
```
Comment (masters3d) [inline exercises/ocr-numbers/OcrNumbersTest.swift]: Do you need two catches here? :)

File: exercises/ocr-numbers/OcrNumbersTest.swift (hunk 1 above)
Comment (masters3d) [inline exercises/ocr-numbers/OcrNumbersTest.swift]: Maybe I am missing something about catching.

Comment (masters3d) [inline exercises/ocr-numbers/OcrNumbersTest.swift]: I get it, you are catching any other error. Mmm. Let's go for it.

**Comment by masters3d** [comment]
Thanks. Looks good. I wish swift error catching was a little more elegant.

File: exercises/ocr-numbers/OcrNumbersTest.swift (hunk 1 above)
Comment (robtimp) [inline exercises/ocr-numbers/OcrNumbersTest.swift]: Yep, `catch` statements have to be exhaustive.

**Pull request merged**